qiskit/transpiler/passes/synthesis/high_level_synthesis.py

```diff
@@ -642,8 +657,9 @@ def _definitely_skip_node(self, node: DAGOpNode, qubits: tuple[int] | None) -> b
             # The fast path is just for Rust-space standard gates (which excludes
             # `AnnotatedOperation`).
             node.is_standard_gate()
-            # If it's a controlled gate, we might choose to do funny things to it.
-            and not node.is_controlled_gate()
+            # At the moment we don't consider fast-path handling for controlled gates over 3 or
+            # more controlled qubits. However, we should not abort early for CX/CZ/etc.
+            and not (node.is_controlled_gate() and node.num_qubits >= 3)
```

Comment:
I think maybe this comment has too many negatives in it?  I think we _do_ abort early for CX/CZ, right?

Or maybe the comment is referring to "abort `_definitely_skip`" and I'm reading it as "abort synthesis for this gate", in which case we might want to tweak the words, since it wasn't clear to me.

Reply:
I have reworded the comment. I have indeed meant that the function ``_definitely_skip`` should not immediately return ``False`` when seeing a CX-gate or a CZ-gate, even though these are controlled gates.